src/docfx/template/liquid/LiquidTemplate.cs

```diff
@@ -39,7 +39,26 @@ public string Render(string templateName, JObject model)
         {
             var template = _templates.GetOrAdd(
                 templateName,
-                new Lazy<Template>(() => LoadTemplate(Path.Combine(_templateDir, templateName + ".html.liquid")))).Value;
+                new Lazy<Template>(() =>
+                {
+                    var fileName = $"{templateName}.html.liquid";
+                    try
+                    {
+                        return LoadTemplate(Path.Combine(_templateDir, fileName));
+                    }
+                    catch (Exception ex)
```

Comment:
Consider use return value to show that template is optional, exceptions are fragile and implicit. If we modify `LoadTemplate` method to read from git, this wrapping logic will fail to function because git might throw a different exception type and people will forget to change here.

Reply:
you are right